app/commands/exercise/authorship/create.rb

```diff
@@ -0,0 +1,21 @@
+class Exercise
+  class Authorship
+    class Create
+      include Mandate
+
+      initialize_with :exercise, :author
+
+      def call
+        authorship = exercise.authorships.create!(author: author)
+        User::ReputationAcquisition.find_or_create_by!(
+          user: author,
+          reason_object: authorship,
+          reason: "exercise_authorship",
+          category: "exercise_authorship"
+        )
+      rescue ActiveRecord::RecordNotUnique
```

Comment:
I reckon it's safer (and clearer) to wrap L9 in this rescue, and return in it. As we could end up with a situation where something inside `User::ReputationAcquisition. find_or_create_by` causes the exception, and then we'd be in a muddled state but not know.

Reply:
Agreed